app/src/main/java/com/coincollection/CollectionPage.java

```diff
@@ -325,6 +344,20 @@ public boolean onCreateOptionsMenu(Menu menu) {
             //saveItem.setVisible(false);
         }
 
+        // Update the filter button text based on current filter state
+        MenuItem filterItem = menu.findItem(R.id.toggle_coin_filter);
+        switch (mCoinFilter) {
```

Comment:
This should show the next state instead of the current state, since when a user selects it, it will take them to the next display
```suggestion
        switch ((mCoinFilter + 1) % 3) {
```